src/loading2/point-attributes.ts

```diff
@@ -47,7 +47,8 @@ class PointAttribute {
 		public name: string,
 		public type: PointAttributeTypeType,
 		public numElements: number,
-		public range: RangeType = [Infinity, -Infinity]
+		public range: RangeType = [Infinity, -Infinity],
+		public uri: string = ""
```

Comment:
> could you add a comment here that describes what this uri is for?

The URI points to the binary buffer on S3 that contains the respective attribute. For normal potree v2 datasets this will be always `octree.bin` where point positions and colors are stored interleaved in a single file. 

For GLTF encoded datasets that are generated by current cloud processing pipelines there are dedicated buffers for positions and colors named `position.glbin` and `colors.glbin` respectively. These are default names and spit out when processing in the cloud. 

However, recently guys from matic requested to be able to define arbitrary S3 locations rather than relying on such implicit naming convention. Therefore the upcoming version of the gltf converter lib will additionally write the reference path to the attribute buffer as stated in the source glTF file. 

So for stuff processed from Matic this can be anything like `123-xyz.glbin` for the positions for instance and the loader still needs to understand how to request that particular attribute buffer.

Reply:
Thank you for this extensive description. Could you add a comment in the code how other potential users may use this uri for their advantage?

Reply:
So for now we utilize internal tooling for generating this particular kind of dataset to support opf gltf pointclouds. Other users will most likely generate their datasets with the stock PotreeConverter 2. In that case I don't see any real benefit for using custom buffer URIs because there is a clear naming convention in the spec for potree format 2.